src/ViewInstance.js

```diff
@@ -556,38 +558,42 @@ class ViewInstance extends React.Component {
         {
           (canCreateMARCHoldingsForInstanceWithSourceMARC || canEditDeriveMARCRecord) && (
             <MenuSection label={intl.formatMessage({ id: 'ui-inventory.quickMARC.label' })} id="quickmarc-menu-section">
-              <IfPermission perm="records-editor.records.item.put">
-                <Button
-                  id="edit-instance-marc"
-                  buttonStyle="dropdownItem"
-                  disabled={!marcRecord}
-                  onClick={() => {
-                    onToggle();
-                    this.editInstanceMarc();
-                  }}
-                >
-                  <Icon icon="edit">
-                    <FormattedMessage id="ui-inventory.editInstanceMarc" />
-                  </Icon>
-                </Button>
-              </IfPermission>
-
-              <IfPermission perm="records-editor.records.item.post">
-                <Button
-                  id="duplicate-instance-marc"
-                  buttonStyle="dropdownItem"
-                  disabled={!marcRecord}
-                  onClick={() => {
-                    onToggle();
-                    this.duplicateInstanceMarc();
-                  }}
-                >
-                  <Icon icon="duplicate">
-                    <FormattedMessage id="ui-inventory.duplicateInstanceMarc" />
-                  </Icon>
-                </Button>
-              </IfPermission>
-
+              {canEditMARC && (
+              <>
+                <IfPermission perm="records-editor.records.item.put">
+                  <Button
+                    id="edit-instance-marc"
+                    buttonStyle="dropdownItem"
+                    disabled={!marcRecord}
+                    onClick={() => {
+                      onToggle();
+                      this.editInstanceMarc();
+                    }}
+                  >
+                    <Icon icon="edit">
+                      <FormattedMessage id="ui-inventory.editInstanceMarc" />
+                    </Icon>
+                  </Button>
+                </IfPermission>
+
+                <IfPermission perm="records-editor.records.item.post">
+                  <Button
+                    id="duplicate-instance-marc"
+                    buttonStyle="dropdownItem"
+                    disabled={!marcRecord}
+                    onClick={() => {
+                      onToggle();
+                      this.duplicateInstanceMarc();
+                    }}
+                  >
+                    <Icon icon="duplicate">
+                      <FormattedMessage id="ui-inventory.duplicateInstanceMarc" />
+                    </Icon>
+                  </Button>
+                </IfPermission>
+              </>
+              )}
+              {canViewMARC && (
               <IfPermission perm="ui-quick-marc.quick-marc-holdings-editor.create">
```

Comment:
This button is to create MARC Holdings, button for viewing source should be somewhere else